libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -733,7 +734,6 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
         IotListDouble_InsertHead( &( mqttConnection->pendingResponse ),
                                   &( pSubscriptionOperation->link ) );
 
-        IotMutex_Unlock( &( mqttConnection->referencesMutex ) );
 
         /* Processing operation after sending it on the network. */
         _IotMqtt_ProcessOperation( pSubscriptionOperation );
```

Comment:
This function has internally mutex lock and unlock on the same reference mutex. So this change may result in a deadlock.

Reply:
fixed in 6894004